File: apps/api/payments/pricing.py
from decimal import Decimal, ROUND_HALF_UP

BASE_FEE_USD = Decimal("2500")
SUCCESS_FEE_RATE = Decimal("0.10")


def compute_success_fee_cents(confirmed_recovery_usd: Decimal) -> int:
    fee = (confirmed_recovery_usd * SUCCESS_FEE_RATE * Decimal("100")).quantize(
        Decimal("1"), rounding=ROUND_HALF_UP
    )
    return int(fee)


def compute_total_amount_usd(confirmed_recovery_usd: Decimal) -> float:
    return float(BASE_FEE_USD + confirmed_recovery_usd * SUCCESS_FEE_RATE)

File: packages/shared/src/pricing.ts
export const VERIFICATION_REPORT_BASE_FEE_USD = 2500;
export const SUCCESS_FEE_RATE = 0.1;

export interface CheckoutTotalBreakdown {
  baseFeeUsd: number;
  successFeeUsd: number;
  totalUsd: number;
}

export function computeCheckoutTotal(confirmedRecoveryUsd: number): CheckoutTotalBreakdown {
  const baseFeeUsd = VERIFICATION_REPORT_BASE_FEE_USD;
  const successFeeUsd = confirmedRecoveryUsd * SUCCESS_FEE_RATE;
  return {
    baseFeeUsd,
    successFeeUsd,
    totalUsd: baseFeeUsd + successFeeUsd,
  };
}

export function computeSuccessFeeCents(confirmedRecoveryUsd: number): number {
  return Math.round(confirmedRecoveryUsd * SUCCESS_FEE_RATE * 100);
}

export function parseUsdAmount(value: string | number): number {
  if (typeof value === "number") {
    return Number.isFinite(value) ? Math.max(0, value) : 0;
  }
  const parsed = Number.parseFloat(String(value).replace(/[^0-9.-]/g, ""));
  return Number.isFinite(parsed) ? Math.max(0, parsed) : 0;
}
